vault/dendron.topic.special-notes.commands.create-journal-note.md

```diff
@@ -0,0 +1,21 @@
+---
+id: yddujnt9s4glcudcwczdobl
+title: "Dendron: Create Journal Note"
+desc: Create a journal note.
+updated: 1661498393820
+created: 1661169220970
+---
+
+## Summary
+
+{{fm.desc}}
+
+## Keybindings
+- windows: `ctrl+shift+j`
+- mac: `cmd+shift+j`
+
+## Details
+
+**{{fm.title}}** is a command that lets you create a [[dendron.topic.special-notes.journal-note]].
+
```

Comment:
I don't think we need to go that lean with the information it is providing. Every other command note has a details section starting with a sentence that tells you what it does in simple english, and without this sentence this note will stick out.

The top level outline of a command is:
- Summary: a one-liner description in simple english on what it does
- keybindings: keybindings if necessary
- details: everything else

This sentence being under details, there will be restatements happening frequently.
Right now this note really doesn't have a lot more detail so this sentence kind of sticks out to look redundant, but to preserve the completeness (in a sense that you can expect the same structure of information organized in every single note about a particular command), I would prefer this sentence to still be here.

Reply:
In short I'm arguing for redundancy for the sake of conformity in the standard we are applying. 

The content is sound without it, but our goal here is provide an expectation that any piece of information you get will be structured in a predictable way. I would say removing it will go against why we are doing this in the first place.

Reply:
Since this has been a common topic for discussion between us, let me go into my reasoning a bit more:

My thoughts on the argument for KISS / removing redundancy in documents.

The argument against keeping this sentence: 

- Imposing this structure of restating what the command does in the `details` section is unnecessary fluff, and an overhead that adds to document maintenance in the future.

The argument for keeping this sentence:

- That may be the case, and it would be beneficial if we are keeping this internally. 
- However, this is a public facing document.
  - With the assumption that readers want clear and predictable documentation, redundancy in documentation is not a concern.
  - In fact, I believe for the sake of keeping public facing documents presentable, the _writers should be the ones who bear the overhead if that is what the standard asks for_.
  - Let's say then, that as a standard that we make this sentence optional.
    - This also comes with its own overhead: _determining if this sentence should go here_.
    - By making it explicitly part of every single command note, we are eliminating this overhead of dealing with a tabula rasa of the `details` section, by suggesting that you start every command note details section with a longer-form description of what this command does.
- Standards by nature introduce redundancy.
  - Every piece of information we document can be reduced so that it contains all the necessary bits of information in the least amount of characters.
  - With standards, we are not doing that. 
    - We are not vacuum-packing information as air-tight as possible for storage, but rather packaging them in a nice box to be on a front window display.
- Bytes are cheap.
  - The redundancy has its purpose, and we are paying very little for that.

Reply:
so i think we should keep the details section. i'm saying we should cut the first sentence because it doesn't add new information. 

in terms of whats expensive, its not the bytes that are expensive but attention and time to read docs that don't add new information. (this was the big problem with javadocs - lots of text, but very little insight)

Reply:
I was under the impression that the template I created (which had the sentence) for past standardization are good to go.
My intention in the template was that each detail section starts with a sentence that reiterates what the command does;
preferably with a bit more meat to it so that it can provide a natural segue into what we are going to talk about in detail.

In the case of these commands there's not much to add for that piece of sentence, but I still value it being there because it serves as a common grounding point to every command note's detail section that you can later expand on.

We have came to a good agreement on the overall structure of each type of note, and here I'm trying to introduce some structure to each block of text so that it serves as some starting point for others when writing each section of a particular note consistently.

Well that was my intention, but I think we need more discussion on implementing that to the workflow, and that shouldn't block this PR.

For now (and in the future unless further discussions change this), I'll go with the following:

- If the command is so simple that the summary is enough to capture what it does, omit the detail section altogether.
- If there are things to add in the detail section, start the detail section with the opening sentence.

Reply:
I know we're constantly iterating on the standards as we go, and I apologize that the reviews turn into lengthy discussions sometimes.

Though without initiating conversation on these frictions during the review we're not going to end up with a solid standard at the end, and the workflow is going to incrementally become chaotic with each new ad-hoc decisions in the process.

I'll try to keep it short and come to a middle ground faster to make this process shorter, and async the conversations so they don't block

Reply:
> for the detail section, we can do a note reference to the detail section of the special note to explain more what this is doing

I'm not sure if that helps with

> (this was the big problem with javadocs - lots of text, but very little insight)

I would very much prefer to keep the details described in a command note to be directly related to the command and not go into the details of the output it creates which is already described elsewhere.

I'll omit the `## Details` section altogether if it makes no sense to have it.